src/XrdPfc/XrdPfcFile.cc

```diff
@@ -820,13 +832,26 @@ int File::ReadOpusCoalescere(IO *io, const XrdOucIOVec *readV, int readVnum,
             {
                TRACEF(DumpXL, tpfx << "direct block " << block_idx << ", blk_off " << blk_off << ", size " << size);
 
-               if (lbe == LB_direct)
-                  iovec_direct.back().size += size;
-               else
-                  iovec_direct.push_back( { block_idx * m_block_size + blk_off, size, 0, iUserBuff + off } );
                iovec_direct_total += size;
                read_req->m_direct_done = false;
 
+               // Make sure we do not issue a ReadV with chunk size above XrdProto::maxRVdsz.
+               // Number of actual ReadVs issued so as to not exceed the XrdProto::maxRvecsz limit
+               // is determined in the RequestBlocksDirect().
+               if (lbe == LB_direct && iovec_direct.back().size + size <= XrdProto::maxRVdsz) {
```

Comment:
Looks like the variable `XrdProto::maxRVdsz` is undefined? At least i was not able to find its definition.

Probably something like `XrdXrootdProtocol::maxReadv_ior` should be here (though it is protected...).

Reply:
This was defined in 5dff60cdd4a8b5ec62c52a9dc0de1590417ed6c6.